### DISCUSSION TITLE
Some tips on adding a falcon group tag based on a hostname prefix and OU

### DISCUSSION THREAD

**Discussion by codechunt** [CrowdStrike/psfalcon#328] [Q&A] [answered]
Im new to PSFalcon. I am working through getting proper FalconGroupTags added to a series of hosts.

Critieria im working with:

Hostname prefix - since its a few thousand hosts with some common attributes is there a way to use Get-Falconhost to pull a list of hosts with a common hostname prefix?

Taking it a step further I'm having difficulty figuring out how to filter with Get-Falconhost based on OU.

**Comment by bk-cs** [comment, marked as answer]
Hostname prefix:
```
Get-FalconHost -Filter "hostname:'prefix'" -All
```
By OU:
```
Get-FalconHost -Filter "ou:'ou name'" -All
```

**Reply by codechunt** [reply]
And If I wanted to supply the Get-FalconHost -Filter "hostname:'prefix'" -All 
with a list of different prefixes?

**Reply by bk-cs** [reply]
Assuming that you mean you want a single list containing any device that matches one of the hostname prefixes, you can repeat up to 20 times using comma (equivalent to "or"):
```powershell
Get-FalconHost -Filter "hostname:'prefix',hostname:'prefix',hostname:'prefix'"
```
See [Filtering Results](https://github.com/CrowdStrike/psfalcon/wiki/Filtering-Results) for more detail.

Because this comes up a lot, I created [Find-FalconHostname](https://github.com/CrowdStrike/psfalcon/wiki/Find-FalconHostname). It defaults to an exact match, so you'll need to use it with the `-Partial` switch to find a prefix. You can supply as many partial hostnames as you'd like, and it will group them by 20:
```powershell
'prefix','prefix','prefix','prefix' | Find-FalconHostname -Partial
```
